Findscript.cmake.in

```diff
@@ -75,58 +75,54 @@ function(getNumBits _ret)
 endfunction()
 ################################################################################
 # TRICKY: clear cached variables each time we cmake so we can change
-# ovsrpro_REV and reuse the same build directory
-unset(ovsrpro_DIR CACHE)
+# @CMAKE_PROJECT_NAME@_REV and reuse the same build directory
+unset(@CMAKE_PROJECT_NAME@_DIR CACHE)
 ################################################################################
-# find the path to the ovsrpro directory
+# find the path to the @CMAKE_PROJECT_NAME@ directory
 getCompilerPrefix(COMPILER)
 getNumBits(BITS)
-set(ovsrpro_SIG ${ovsrpro_REV}-${COMPILER}-${BITS})
+set(@CMAKE_PROJECT_NAME@_SIG ${@CMAKE_PROJECT_NAME@_REV}-${COMPILER}-${BITS})
 set(PFX86 "ProgramFiles(x86)")
-find_path(ovsrpro_DIR
+find_path(@CMAKE_PROJECT_NAME@_DIR
   NAMES
-    ovsrpro_${ovsrpro_SIG}.txt
+    @CMAKE_PROJECT_NAME@_${@CMAKE_PROJECT_NAME@_SIG}.txt
   PATHS
     # build versions
-    C:/src/ovsrpro/_bld/ovsrpro_${ovsrpro_SIG}
-    ~/src/ovsrpro/_bld/ovsrpro_${ovsrpro_SIG}
+    C:/src/@CMAKE_PROJECT_NAME@/_bld/@CMAKE_PROJECT_NAME@_${@CMAKE_PROJECT_NAME@_SIG}
+    ~/src/@CMAKE_PROJECT_NAME@/_bld/@CMAKE_PROJECT_NAME@_${@CMAKE_PROJECT_NAME@_SIG}
     # environment variable
-    "$ENV{ovsrpro}/ovsrpro ${ovsrpro_SIG}"
-    "$ENV{ovsrpro_DIR}/ovsrpro-${ovsrpro_SIG}-${CMAKE_SYSTEM_NAME}"
+    "$ENV{ovsrpro_DIR}/@CMAKE_PROJECT_NAME@ ${@CMAKE_PROJECT_NAME@_SIG}"
+    "$ENV{ovsrpro_DIR}/@CMAKE_PROJECT_NAME@-${@CMAKE_PROJECT_NAME@_SIG}-${CMAKE_SYSTEM_NAME}"
     # installed versions
-    "$ENV{ProgramW6432}/ovsrpro ${ovsrpro_SIG}"
-    "$ENV{${PFX86}}/ovsrpro ${ovsrpro_SIG}"
-    "~/ovsrpro/ovsrpro-${ovsrpro_SIG}-${CMAKE_SYSTEM_NAME}"
-    "/opt/ovsrpro/ovsrpro-${ovsrpro_SIG}-${CMAKE_SYSTEM_NAME}"
-    # symbolic link install option
-    "/opt/ovsrpro/ovsrpro"
-    # rpm installed location
-    "/opt/ovsrpro"
-  DOC "ovsrpro directory"
+    "$ENV{ProgramW6432}/@CMAKE_PROJECT_NAME@ ${@CMAKE_PROJECT_NAME@_SIG}"
+    "$ENV{${PFX86}}/@CMAKE_PROJECT_NAME@ ${@CMAKE_PROJECT_NAME@_SIG}"
+    "~/ovsrpro/@CMAKE_PROJECT_NAME@-${@CMAKE_PROJECT_NAME@_SIG}-${CMAKE_SYSTEM_NAME}"
+    "/opt/ovsrpro/@CMAKE_PROJECT_NAME@-${@CMAKE_PROJECT_NAME@_SIG}-${CMAKE_SYSTEM_NAME}"
```

Comment:
does the install path on UNIX systems need to be `~/ovsrpro` and `/opt/ovsrpro`? the recent change in externpro to make this `~/extern` and `/opt/extern` was meant to be a more generic place to install - and a single mount point on our shared development systems (install once on a single system and have externpro/internpro/webpro and skylynx pro-like installs all in `/opt/extern`)

Reply:
@smanders that would be helpful. it will of course cause some downstream changes but that would be excellent. do we just need to remove this Findscript file?

Reply:
before we remove it I want to make sure the externpro `Findscript.cmake.in` has everything you want/need

Reply:
@smanders we were also using the old Findovspro.cmake file just like the externpro was used by code that included this stuff. if we don't need the find script anymore, that is fine by me.

Reply:
Maybe there’s just confusion on what the Find*pro.cmake script is for...

* externpro does not have a Findexternpro.cmake file in it’s repo
* the Findexternpro.cmake is generated at cmake-time from the Findscript.cmake.in template
* the generated Findexternpro.cmake file is in the externpro install (share/cmake)
* this generated file is copied to projects that need to find externpro
* the project finds externpro via this file and compares it to the Findexternpro.cmake that’s in the release of externpro that it found — if they’re different then the developer gets a cmake warning that the files don’t match and the updated version from externpro should be copied to the project

bottom line is that you shouldn’t need to have a Findovsrpro.cmake in the ovsrpro repo - it will be generated at cmake-time and will be part of the install (in share/cmake)

and if the Findscript.cmake.in template (from externpro) doesn’t have something you need in your Findovsrpro.cmake then we should consider modifying Findscript.cmake.in in externpro...

Reply:
@smanders thanks for the explanation. that helps a lot. I will remove the find script from here.

Reply:
@smanders is there a macro i need to call to ensure the findovspro script is generated?

Reply:
@swflb no, you shouldn’t need to call a macro - it should be done for you...
```bash
$ cmake -DXP_STEP=build /path/to/src
```
look in staging directory share/cmake (for example internpro_18.01.2-gcc630-64/share/cmake/ in the build directory for internpro) and you should see a Find*pro.cmake (Findinternpro.cmake in internpro)

how this happens:
* these lines are called from `proSetStageDir()` macro https://github.com/smanders/externpro/blob/18.01.1/modules/macpro.cmake#L292-L295
* the `proSetStageDir()` macro is called from the cmake-generated pro_build.cmake file that’s in your build directory https://github.com/smanders/externpro/blob/18.01.1/modules/macpro.cmake#L245